src/hmr/hotModuleReplacement.js

```diff
@@ -168,6 +176,13 @@ function reloadAll() {
   });
 }
 
+function isDataURL(s) {
+  return !!s.match(
+    // eslint-disable-next-line no-useless-escape
+    /^\s*data:([a-z]+\/[a-z0-9-+.]+(;[a-z-]+=[a-z0-9-]+)?)?(;base64)?,([a-z0-9!$&',()*+;=\-._~:@\/?%\s]*)\s*$/i
```

Comment:
oh, let's simplify this, no need long regexp, they decrease perf.

Example what we use https://github.com/webpack/loader-utils/blob/master/lib/isUrlRequest.js#L9

https://github.com/webpack/loader-utils/blob/master/lib/isUrlRequest.js#L14

Because we can have `#hash`, `file:`, `http:`/`https:` (absolute urls), schema less urls `//something.com` and `chrome-extension:`, i think we should ignore their too

I can accept this without tests, but will be great have tests for this to avoid future regressions

Reply:
So could a modified version like the following be acceptable?
```function isUrlRequest(url) {
  // An URL is not an request if

  if (/^[a-z][a-z0-9+.-]*:/i.test(url)) {
    return false;
  }

  // 2. It's a protocol-relative
  if (/^\/\//.test(url)) {
    return false;
  }

  // 3. It's some kind of url for a template
  if (/^[{}[\]#*;,'§$%&(=?`´^°<>]/.test(url)) {
    return false;
  }

  // 4. It's also not an request if root isn't set and it's a root-relative url
  if (/^\//.test(url)) {
    return false;
  }

  return true;
}
```

Reply:
No need `4` and for `3` only `#` character

Reply:
So this? :) 
```function isUrlRequest(url) {
  // An URL is not an request if

  // 1. It's an absolute url
  if (/^[a-z][a-z0-9+.-]*:/i.test(url)) {
    return false;
  }

  // 2. It's a protocol-relative
  if (/^\/\//.test(url)) {
    return false;
  }

  return true;
}```

Reply:
and :
```js
// 3. Avoid `#`
if (/^#/.test(url)) {
  return false;
}
```

To avoid `link="#test"` (based on spec it is allowed).